### ISSUE TITLE
some questions about training epoch

### ISSUE THREAD

**Issue by muyouyang** [xuanqing94/AdvGAN#4]
can you tell me the training epoch number before finetuning for imagenet 143 category and cifar datasets

**Comment by xuanqing94** [comment]
I remembered 10-20 epochs are just fine. But you can always wait until the accuracy does not improve.

**Comment by muyouyang** [comment]
Do you mean train 10-20 epoch for imagenet 143 category ?what about finetune epoch? i found that after finetune, the  acc_fake of discrimiantor in test dataset is high and the acc_real  around 0.85 is very low only around 0.03. Is there something wrong with my training?

**Comment by muyouyang** [comment]
also  if found that you mention that acc under attack needs to do be done. the test_acc in your finetune.py can not be regard as your acc under attack? 
i don't know why my test result only around 0.03, thanks for your explanation.

**Comment by muyouyang** [comment]
also i found that the fintune _dis_real_loss can not converage

**Comment by xuanqing94** [comment]
> Do you mean train 10-20 epoch for imagenet 143 category ?what about finetune epoch? i found that after finetune, the acc_fake of discrimiantor in test dataset is high and the acc_real around 0.85 is very low only around 0.03. Is there something wrong with my training?

What do you mean? acc_real is only 0.03?

**Comment by muyouyang** [comment]
<img width="346" alt="Screen Shot 2019-12-17 at 09 50 30" src="https://user-images.githubusercontent.com/32900221/70958124-2cbc6380-20b3-11ea-8b34-e7ab702066a0.png">
 this is the finetune.txt

**Comment by xuanqing94** [comment]
What's your lambda value? It seems that you are not using the real data to train. And is it adversarial training?

**Comment by muyouyang** [comment]
<img width="618" alt="Screen Shot 2019-12-17 at 23 22 40" src="https://user-images.githubusercontent.com/32900221/71008938-6e392700-2124-11ea-8e19-a7f9a6b1672a.png">
 i found in your training code, i once thought you intentionally only training adversarial examples. Do you mean add this in train.py. ori means real picture without adversarial training
<img width="871" alt="Screen Shot 2019-12-17 at 23 44 17" src="https://user-images.githubusercontent.com/32900221/71013959-1c949a80-212c-11ea-9e73-455f851ad123.png">
<img width="414" alt="Screen Shot 2019-12-17 at 23 44 13" src="https://user-images.githubusercontent.com/32900221/71013962-1d2d3100-212c-11ea-90da-f0ebbcceb82d.png">

**Comment by xuanqing94** [comment]
But this is the co-training part (step 1), not fine-tuning. For # training epochs required in ImageNet-143 data, you can refer to Figure 8 in our paper. It takes ~60 epochs in total.

**Comment by muyouyang** [comment]
i feel sorry for  my understanding.  i have read your readme file.
1. con train with (real_adv+fake)
2.finetune(real_adv+fake_adv)
does this two step includes all of your train and fintune process?  I cannot figure out what did you mean finetune for real dat and when does it happen?

**Comment by muyouyang** [comment]
My understanding about your adversarial training is your first step co-train, and data augmentation is your second step finetune using fake_adv ?i

**Comment by xuanqing94** [comment]
> i feel sorry for my understanding. i have read your readme file.
> 
> 1. con train with (real_adv+fake)

Co-train is to train a good GAN model

>    2.finetune(real_adv+fake_adv)

Finetune is purely for robust classification. We augment the training set with fake data.

>    does this two step includes all of your train and fintune process?  I cannot figure out what did you mean finetune for real dat and when does it happen?

Yes. Finetune on real data is just like how you train a normal classifier, in this case, we only need to keep the classification branch of the discriminator.
For more information, I recommend you to read the paper and unofficial blogs (in Chinese) (https://blog.csdn.net/waple_0820/article/details/99983510, https://tiantianwahaha.github.io/2019/07/24/Rob-GAN-Generator-Discriminator-and-Adversarial-Attacker/)

**Comment by 1996-jb** [comment]
This RobGan is a defence model?
